src/cloudflare/internal/ai-api.ts

```diff
@@ -221,6 +245,82 @@ export class Ai {
     }
   }
 
+  public async toMarkdown(
+    files: { name: string; blob: Blob }[],
+    options?: { gateway?: GatewayOptions }
+  ): Promise<ConversionResponse[]>;
+  public async toMarkdown(
+    files: {
+      name: string;
+      blob: Blob;
+    },
+    options?: { gateway?: GatewayOptions }
+  ): Promise<ConversionResponse>;
+  public async toMarkdown(
+    files: { name: string; blob: Blob } | { name: string; blob: Blob }[],
+    options?: { gateway?: GatewayOptions }
+  ): Promise<ConversionResponse | ConversionResponse[]> {
+    const input = Array.isArray(files) ? files : [files];
+
+    const processedFiles = [];
+    for (const file of input) {
+      processedFiles.push({
+        name: file.name,
+        mimeType: file.blob.type,
+        data: await blobToBase64(file.blob),
+      });
+    }
+
+    const fetchOptions = {
+      method: 'POST',
+      body: JSON.stringify({
+        files: processedFiles,
+        options: options,
+      }),
+      headers: {
+        'content-type': 'application/json',
+      },
+    };
+
+    const endpointUrl =
+      'https://workers-binding.ai/to-everything/markdown/transformer';
+
+    const res = await this.fetcher.fetch(endpointUrl, fetchOptions);
+
+    switch (res.status) {
+      case 200: {
```

Comment:
A better practice would be to check for `if (res.ok) { ...` which covers the full range of 2xx status codes, not just 200, but this is fine if you know your service would only ever produce 200 codes.

Reply:
this service only produces 200
I used the switch statement, because other methods in this file produce several codes and use it already